letsencrypt/cli.py

```diff
@@ -739,62 +749,66 @@ def add_subparser(name, func):  # pylint: disable=missing-docstring
 
     # the order of add_subparser() calls is important: it defines the
     # order in which subparser names will be displayed in --help
+    # these add_subparser objects return objects to which arguments could be
+    # attached, but they have annoying arg ordering constrains so we use
+    # groups instead: https://github.com/letsencrypt/letsencrypt/issues/820
+
     add_subparser("run", run)
-    parser_auth = add_subparser("auth", auth)
-    parser_install = add_subparser("install", install)
-    parser_revoke = add_subparser("revoke", revoke)
-    parser_rollback = add_subparser("rollback", rollback)
+    add_subparser("auth", auth)
+    helpful.add_group("auth", description="Options for modifying how a cert is obtained")
+    add_subparser("install", install)
+    helpful.add_group("install", description="Options for modifying how a cert is deployed")
+    add_subparser("revoke", revoke)
+    helpful.add_group("revoke", description="Options for revocation of certs")
+    add_subparser("rollback", rollback)
+    helpful.add_group("rollback", description="Options for reverting config changes")
```

Comment:
With everything moving to argument groups, what is the purpose of the subparsers? To control what function runs?

A bit of a nitpick, but I think like it would be clearer and less code to change `VERBS` to a `dict` that maps verbs to functions. You could still iterate over `VERBS` the same way and lookup the desired function in the bottom of `main`.